File: .NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/English/TimexConstantsEnglish.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;

namespace Microsoft.Recognizers.Text.DataTypes.TimexExpression
{
    internal static class TimexConstantsEnglish
    {
        public static readonly string[] Days =
        {
            "Monday",
            "Tuesday",
            "Wednesday",
            "Thursday",
            "Friday",
            "Saturday",
            "Sunday",
        };

        public static readonly string[] Months =
        {
            "January",
            "February",
            "March",
            "April",
            "May",
            "June",
            "July",
            "August",
            "September",
            "October",
            "November",
            "December",
        };

        public static readonly string[] DateAbbreviation =
        {
            "th",
            "st",
            "nd",
            "rd",
            "th",
            "th",
            "th",
            "th",
            "th",
            "th",
        };

        public static readonly string[] Hours =
        {
            "midnight", "1AM", "2AM", "3AM", "4AM", "5AM", "6AM", "7AM", "8AM", "9AM", "10AM", "11AM",
            "midday", "1PM", "2PM", "3PM", "4PM", "5PM", "6PM", "7PM", "8PM", "9PM", "10PM", "11PM",
        };

        public static readonly IDictionary<string, string> Seasons = new Dictionary<string, string>
        {
            { "SP", "spring" },
            { "SU", "summer" },
            { "FA", "fall" },
            { "WI", "winter" },
        };

        public static readonly string[] Weeks =
        {
            "first",
            "second",
            "third",
            "forth",
        };

        public static readonly IDictionary<string, string> DayParts = new Dictionary<string, string>
        {
            { "DT", "daytime" },
            { "NI", "night" },
            { "MO", "morning" },
            { "AF", "afternoon" },
            { "EV", "evening" },
        };
    }
}

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/english/TimexConstantsEnglish.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.datatypes.timex.expression.english;

import java.util.HashMap;
import java.util.Map;

public class TimexConstantsEnglish {
    public static final String EVERY = "every";
    public static final String NOW = "now";
    public static final String MIDNIGHT = "midnight";
    public static final String MIDDAY = "midday";
    public static final String TODAY = "today";
    public static final String TOMORROW = "tomorrow";
    public static final String YESTERDAY = "yesterday";
    public static final String WEEKEND = "weekend";
    public static final String TONIGHT = "tonight";
    public static final String THIS = "this";
    public static final String LAST = "last";
    public static final String NEXT = "next";

    public static final String[] DAYS = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday",
        "Sunday" };

    public static final String[] MONTHS = { "January", "February", "March", "April", "May", "June", "July", "August",
        "September", "October", "November", "December", };

    public static final String[] DATE_ABBREVIATION = { "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th", };

    public static final String[] HOURS = { "midnight", "1AM", "2AM", "3AM", "4AM", "5AM", "6AM", "7AM", "8AM", "9AM",
        "10AM", "11AM", "midday", "1PM", "2PM", "3PM", "4PM", "5PM", "6PM", "7PM", "8PM", "9PM", "10PM", "11PM", };

    public static final Map<String, String> SEASONS = new HashMap<String, String>() {
        {
            put("SP", "spring");
            put("SU", "summer");
            put("FA", "fall");
            put("WI", "winter");
        }
    };

    public static final String[] WEEKS = { "first", "second", "third", "forth", };

    public static final Map<String, String> DAY_PARTS = new HashMap<String, String>() {
        {
            put("DT", "daytime");
            put("NI", "night");
            put("MO", "morning");
            put("AF", "afternoon");
            put("EV", "evening");
        }
    };
}
